src/net/packet.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020 The reone project contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace reone {
+
+namespace net {
+
+bool isBigEndian();
+void unitTestSetEndian(bool isBig);
+
+template <typename T>
+std::enable_if_t< std::is_integral_v<T> || std::is_floating_point_v<T>> swapByte(T &val) {
```

Comment:
Single responsibility principle. "common/endianutil" should be extended as needed and used here instead.